torch/_prims_common/wrappers.py

```diff
@@ -320,12 +319,17 @@ def _fn(*args, out=None, **kwargs):
             sig.empty,
             out_type,
         )
-        params = chain(sig.parameters.values(), (out_param,))
+        params = *sig.parameters.values(), out_param
+
+        # If there's a Parameter.VAR_KEYWORD parameter (like **kwds), it must appear
+        # after the out= parameter, which is Parameter.KEYWORD_ONLY. Sorting by
+        # Parameter.kind guarantees that all the parameters are in legal order.
+        params = sorted(params, key=lambda p: p.kind)
+
         _fn.__signature__ = inspect.Signature(  # type: ignore[attr-defined]
             parameters=params, return_annotation=return_type  # type: ignore[arg-type]
         )
 
-        _fn.__annotations__ = fn.__annotations__
```

Comment:
ah, now you can revert this line. Feel free to merge after you do :)

Reply:
Arg!

Great catch there, Mario!

And the original line isn't right - I fixed it before - it should be:

`_fn.__annotations__ = dict(fn.__annotations__)`

because we go on to mutate that dictionary... :-D :-D

Wilco!